packages/babel-plugin-ember-test-metadata/utils.js

```diff
@@ -56,48 +56,73 @@ function getProjectConfiguration(project) {
+    .replace(`${projectName}/`, '');
 
-  pathSegments.splice(
-    0,
-    pathSegments.indexOf(projectNamePathSeparators[0]) + projectNamePathSeparators.length
-  );
-
-  return pathSegments.join(path.sep);
+  return formattedPath;
 }
 
-function _getParsedEmbroiderFilepath(pathSegments) {
-  const RELATIVE_PATH_ROOT = 2;
+function _getClassicPathWithAddon(preformattedPath, projectName, addons) {
+  const pathSegments = preformattedPath.split(path.sep);
+  const addonNames = addons.paths.map(addonPath => addonPath.split(path.sep)[1]);
+  const addonName = addonNames.find(name => pathSegments.includes(name));
 
-  // A typical Embroider pathSegments array would look something like:
-  // ['private', 'var', 'some-hash', 'embroider', 'another-hash', 'tests', 'acceptance', ...]
-  // This strips everything from the first segment up to and including 'another-hash',
-  pathSegments.splice(0, pathSegments.lastIndexOf('embroider') + RELATIVE_PATH_ROOT);
+  if (pathSegments.includes(addonName)) {
+    pathSegments.splice(0, pathSegments.indexOf("ember-add-in-repo-tests") + 1);
+    return [...pathSegments].join(path.sep)
+  } else {
+    return _getClassicPath(preformattedPath, projectName);
+  }
+}
 
+function _getEmbroiderPath(pathSegments) {
+  pathSegments.splice(0, pathSegments.lastIndexOf('tests'));
   return pathSegments.join(path.sep);
 }
 
+function _getEmbroiderPathWithAddon(pathSegments, addons) {
+  const addonNames = addons.paths.map(addonPath => addonPath.split(path.sep)[1]);
+  const addonName = addonNames.find(name => pathSegments.includes(name));
+
+  if (pathSegments.includes(addonName)) {
+    pathSegments.splice(0, pathSegments.lastIndexOf(addonName) + 1);
+    const hasTestsPrefix = pathSegments[0] === "tests";
+    const addonPrefix = `lib/${addonName}`;
+    return [addonPrefix, hasTestsPrefix ? pathSegments.join('/') : `tests/${pathSegments.join('/')}`].join('/');
+  } else {
+    return _getEmbroiderPath(pathSegments);
+  }
+}
+
 /**
  * Get a normalized file path, based on whether the app build is classic or with Embroider
  * @param {object} fileOpts Babel state.file.opts which include root and filename props
  * @param {object} projectConfiguration Contains project name, ember-addon path info
  * @returns {string} E.g. tests/acceptance/my-test.js
  */
 function getNormalizedFilePath(fileOpts, projectConfiguration) {
-  let { root, filename } = fileOpts;
+  let { filename } = fileOpts;
   const pathSegments = filename.split(path.sep);
   const isEmbroider = pathSegments.includes('embroider');
-
-    filename = isEmbroider ? _getParsedEmbroiderFilepath(pathSegments) : _getParsedClassicFilepath(pathSegments, projectConfiguration);
-
-  return path.relative(root, filename);
+  const addons = projectConfiguration.pkg["ember-addon"];
+  const hasAddons = addons && addons.paths;
+  const projectName = projectConfiguration.pkg.name;
+
+  if (!isEmbroider && !hasAddons) {
```

Comment:
I don't think this logic is correct. It's not about whether the project has addons, but whether the `filename` is from an addon path. The logic here is we need to determine if the app is using in-repo addons, then further testing if the path is either the classic case or the classic in-repo addon case, so this should actually be a nested conditional.

The same applies to those below.